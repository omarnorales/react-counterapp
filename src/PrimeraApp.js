
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ( {saludo, subtitle } ) => {

    if(!saludo){
        throw new Error('Saludo is mandatory');
    }

    return (
        <>
            <h1>{saludo}</h1>
            <p>{subtitle}</p>
            {/* <pre>{ JSON.stringify(objPerson, null, 3) }</pre> */}
        </>
    );

}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    other: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitle: 'I am a subtitle'
}

export default PrimeraApp;