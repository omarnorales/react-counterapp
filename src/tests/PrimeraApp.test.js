import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
// import { render } from '@testing-library/react';

import PrimeraApp from '../PrimeraApp';


describe('Tests on PrimeraApp', () => {

    // test('should show msg: Hello, Im Goku', () => {

    //     const saludo = 'Hello, Im Goku';
    //     const other = 'other';

    //     const { getByText } = render( <PrimeraApp saludo = {saludo} other = {other} />)

    //     expect( getByText(saludo) ).toBeInTheDocument();
    // });

    test('should show <PrimeraApp /> correctamente', () => {

        const saludo = 'Hello, Im Goku';
        const other = 'other';

        const wrapper = shallow(<PrimeraApp saludo = {saludo} other = {other} />);

        expect(wrapper).toMatchSnapshot();

    });

    test('should show subtitle setted by props', () => {

        const saludo = 'Hello, Im Goku';
        const other = 'other';
        const subtitle = "I am a subtitle";

        const wrapper = shallow(
        <PrimeraApp 
            saludo = {saludo} 
            other = {other}
            subtitle = {subtitle}/>
        );

        const textP = wrapper.find('p').text();

        expect(textP).toBe(subtitle);
    })
    
})
