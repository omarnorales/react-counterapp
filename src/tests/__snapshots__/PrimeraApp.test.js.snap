// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on PrimeraApp should show <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hello, Im Goku
  </h1>
  <p>
    I am a subtitle
  </p>
</Fragment>
`;
